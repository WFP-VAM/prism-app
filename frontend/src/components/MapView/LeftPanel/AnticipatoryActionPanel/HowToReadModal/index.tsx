import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
  Typography,
  createStyles,
  makeStyles,
} from '@material-ui/core';
import { Cancel, Close, HelpOutline } from '@material-ui/icons';
import { useSafeTranslation } from 'i18n';
import React from 'react';
import { black, cyanBlue } from 'muiTheme';

const content = [
  { title: 'Window 1', text: 'Start to mid of the rainfall season.' },
  { title: 'Window 2', text: 'Mid to end of the rainfall season.' },
  {
    title: 'Moderate category',
    text: 'Drought events that happen on average 1 in 5 years.',
  },
  {
    title: 'Severe category',
    text: 'Drought events that happen on average 1 in 7 years.',
  },
  {
    title: 'Ready, Set and Go phases',
    text:
      'The “Ready, Set & Go!” system uses seasonal forecasts with longer lead time for preparedness (Ready phase) and shorter lead times for activation and mobilization (Set & Go! phases).',
  },
];

interface HowToReadModalProps {
  open: boolean;
  onClose: () => void;
}

function HowToReadModal({ open, onClose }: HowToReadModalProps) {
  const { t } = useSafeTranslation();
  const classes = useStyles();

  return (
    <Dialog open={open}>
      <DialogTitle>
        <div className={classes.titleWrapper}>
          <Typography variant="h2">{t('How to read this screen')}</Typography>
          <IconButton style={{ padding: 0 }} onClick={() => onClose()}>
            <Cancel />
          </IconButton>
        </div>
      </DialogTitle>
      <DialogContent dividers>
        <div className={classes.contentWrapper}>
          {content.map(x => (
            <div className={classes.contentItem} key={x.title.toLowerCase()}>
              <Typography variant="h3" style={{ fontWeight: 'bold' }}>
                {t(x.title)}
              </Typography>{' '}
              <Typography variant="h3">{t(x.text)}</Typography>{' '}
            </div>
          ))}
        </div>
      </DialogContent>
      <DialogActions className={classes.dialogActionsWrapper}>
        <Button
          component="a"
          href="mailto:anticipatory_action@wfp.org"
          target="_blank"
          className={classes.dialogButton}
          variant="outlined"
          startIcon={<HelpOutline />}
        >
          <Typography>{t('I STILL NEED HELP')}</Typography>
        </Button>
        <Button
          className={classes.dialogButton}
          variant="outlined"
          startIcon={<Close />}
          onClick={() => onClose()}
        >
          <Typography>{t('Close')}</Typography>
        </Button>
      </DialogActions>
    </Dialog>
  );
}

const useStyles = makeStyles(() =>
  createStyles({
    titleWrapper: {
      display: 'flex',
      flexDirection: 'row',
      flexWrap: 'nowrap',
      justifyContent: 'space-between',
    },
    contentWrapper: { display: 'flex', flexDirection: 'column', gap: '2rem' },
    contentItem: { display: 'flex', flexDirection: 'column', gap: '1rem' },
    dialogActionsWrapper: {
      display: 'flex',
      justifyContent: 'space-between',
      padding: '1rem',
    },
    dialogButton: { borderColor: cyanBlue, color: black },
  }),
);

export default HowToReadModal;
