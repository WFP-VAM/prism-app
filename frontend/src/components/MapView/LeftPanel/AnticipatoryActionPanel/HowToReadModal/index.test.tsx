import { render } from '@testing-library/react';
import React from 'react';
import HowToReadModal from '.';

test('renders actions modal', () => {
  const { container } = render(<HowToReadModal open onClose={() => {}} />);
  expect(container).toMatchSnapshot();
});
