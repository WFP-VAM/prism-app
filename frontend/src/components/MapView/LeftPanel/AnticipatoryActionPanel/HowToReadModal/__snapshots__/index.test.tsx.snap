// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders actions modal 1`] = `
<div>
  <mock-dialog
    open="true"
  >
    <mock-dialogtitle>
      <div
        class="makeStyles-titleWrapper-1"
      >
        <mock-typography
          variant="h2"
        >
          How to read this screen
        </mock-typography>
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          style="padding: 0px;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </mock-dialogtitle>
    <mock-dialogcontent
      dividers="true"
    >
      <div
        class="makeStyles-contentWrapper-2"
      >
        <div
          class="makeStyles-contentItem-3"
        >
          <mock-typography
            style="font-weight: bold;"
            variant="h3"
          >
            Window 1
          </mock-typography>
           
          <mock-typography
            variant="h3"
          >
            Start to mid of the rainfall season.
          </mock-typography>
           
        </div>
        <div
          class="makeStyles-contentItem-3"
        >
          <mock-typography
            style="font-weight: bold;"
            variant="h3"
          >
            Window 2
          </mock-typography>
           
          <mock-typography
            variant="h3"
          >
            Mid to end of the rainfall season.
          </mock-typography>
           
        </div>
        <div
          class="makeStyles-contentItem-3"
        >
          <mock-typography
            style="font-weight: bold;"
            variant="h3"
          >
            Mild category
          </mock-typography>
           
          <mock-typography
            variant="h3"
          >
            Drought events that typically occur once every 4 years.
          </mock-typography>
           
        </div>
        <div
          class="makeStyles-contentItem-3"
        >
          <mock-typography
            style="font-weight: bold;"
            variant="h3"
          >
            Moderate category
          </mock-typography>
           
          <mock-typography
            variant="h3"
          >
            Drought events that typically occur once every 5 years.
          </mock-typography>
           
        </div>
        <div
          class="makeStyles-contentItem-3"
        >
          <mock-typography
            style="font-weight: bold;"
            variant="h3"
          >
            Severe category
          </mock-typography>
           
          <mock-typography
            variant="h3"
          >
            Drought events that typically occur once every 7 years.
          </mock-typography>
           
        </div>
        <div
          class="makeStyles-contentItem-3"
        >
          <mock-typography
            style="font-weight: bold;"
            variant="h3"
          >
            Ready, Set and Go phases
          </mock-typography>
           
          <mock-typography
            variant="h3"
          >
            The "Ready, Set & Go!" system uses seasonal forecasts with longer lead time for preparedness (Ready phase) and shorter lead times for activation and mobilization (Set & Go! phases).
          </mock-typography>
           
        </div>
      </div>
    </mock-dialogcontent>
    <mock-dialogactions
      classname="makeStyles-dialogActionsWrapper-4"
    >
      <mock-button
        classname="makeStyles-dialogButton-5"
        component="a"
        href="mailto:anticipatory_action@wfp.org"
        starticon="[object Object]"
        target="_blank"
        variant="outlined"
      >
        <mock-typography>
          I STILL NEED HELP
        </mock-typography>
      </mock-button>
      <mock-button
        classname="makeStyles-dialogButton-5"
        starticon="[object Object]"
        variant="outlined"
      >
        <mock-typography>
          Close
        </mock-typography>
      </mock-button>
    </mock-dialogactions>
  </mock-dialog>
</div>
`;
